stock_landed_costs_delivery/models/purchase_order.py

```diff
@@ -0,0 +1,23 @@
+# Copyright 2021 Tecnativa - Víctor Martínez
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+from odoo import models
+
+
+class PurchaseOrder(models.Model):
+    _inherit = "purchase.order"
+
+    def _prepare_slc_line_values(self):
+        tmpl = self.carrier_id.lc_product_id.product_tmpl_id
+        return {
+            "product_id": self.carrier_id.lc_product_id.id,
+            "name": self.carrier_id.name,
+            "account_id": tmpl.get_product_accounts()["stock_input"].id,
+            "price_unit": self.delivery_price,
+            "split_method": "equal",
+        }
+
+    def _prepare_slc_values(self):
+        res = super()._prepare_slc_values()
+        if self.carrier_id.lc_product_id:
+            res["cost_lines"] = [(0, 0, self._prepare_slc_line_values())]
```

Comment:
Be tolerant with other possible modules adding lines:

```suggestion
            if "cost_lines" not in res:
                res["cost_line"] = []
            res["cost_lines"].append((0, 0, self._prepare_slc_line_values()))
```